system-contracts/contracts/SystemContext.sol

```diff
@@ -74,10 +74,21 @@ contract SystemContext is ISystemContext, ISystemContextDeprecated, ISystemContr
     /// - Their number will start from being equal to the number of the batch and it will increase until it reaches the L2 block number.
     /// - Their timestamp is updated each time a new virtual block is created.
     /// - Their hash is calculated as `keccak256(uint256(number))`
-    BlockInfo internal currentVirtualL2BlockInfo;
+    BlockInfo internal DEPRECATED_currentL2VirtualBlockInfo;
 
     /// @notice The information about the virtual blocks upgrade, which tracks when the migration to the L2 blocks has started and finished.
-    VirtualBlockUpgradeInfo internal virtualBlockUpgradeInfo;
+    struct DEPRECATED_VirtualBlockUpgradeInfo {
+        /// @notice In order to maintain consistent results for `blockhash` requests, we'll
+        /// have to remember the number of the batch when the upgrade to the virtual blocks has been done.
+        /// The hashes for virtual blocks before the upgrade are identical to the hashes of the corresponding batches.
+        uint128 virtualBlockStartBatch;
+        /// @notice L2 block when the virtual blocks have caught up with the L2 blocks. Starting from this block,
+        /// all the information returned to users for block.timestamp/number, etc should be the information about the L2 blocks and
+        /// not virtual blocks.
+        uint128 virtualBlockFinishL2Block;
+    }
+    DEPRECATED_VirtualBlockUpgradeInfo internal virtualBlockUpgradeInfo;
```

Comment:
solidity stores storage variables in packed order: you can delete the `DEPRECATED_VirtualBlockUpgradeInfo` and replace it with uint256 and call the variable deprecated